gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -2234,105 +2328,59 @@ void Movable::AddWayPoint(const Point &Des)
 	while(endNode->Next) {
 		endNode = endNode->Next;
 	}
-	Point p(endNode->x * 16, endNode->y * 12);
+	Point p(endNode->x, endNode->y);
 	area->ClearSearchMapFor(this);
-	PathNode *path2 = area->FindPath( p, Des, size );
+	PathNode *path2 = area->FindPath(p, Des, size);
 	endNode->Next = path2;
 	//probably it is wise to connect it both ways?
 	path2->Parent = endNode;
 }
 
-void Movable::FixPosition()
+// This function is called at each tick if an actor is following another actor
+// Therefore it's rate-limited to avoid actors being stuck as they keep pathfinding
+void Movable::WalkTo(const Point &Des, int distance)
```

Comment:
I suggest you add an 
Actor *actor = nullptr;
if ST_ACTOR ... cast

and then you can use it everywhere without all the repeated casting *and* it removes the need for branching for that first FindPath call.